python/GafferSceneTest/SceneAlgoTest.py

```diff
@@ -1803,6 +1804,123 @@ def assertOptionHistory( optionName, nameSwitchInput, value ) :
 			context["scene:path"] = IECore.InternedStringVectorData( [ "plane" ] )
 			assertOptionHistory( "render:camera", nameSwitch["in"][1]["value"], "/020" )
 
+	def testOptionHistoryWithExpression( self ) :
+
+		script = Gaffer.ScriptNode()
+
+		script["options"] = GafferScene.StandardOptions()
+		script["options"]["options"]["renderCamera"]["enabled"].setValue( True )
+		script["options"]["options"]["renderCamera"]["value"].setValue( "test" )
+
+		script["dot"] = Gaffer.Dot()
+		script["dot"].setup( script["options"]["out"] )
+		script["dot"]["in"].setInput( script["options"]["out"] )
+
+		script["expression"] = Gaffer.Expression()
+		script["expression"].setExpression( inspect.cleandoc(
+		"""
+		globals = parent["dot"]["in"]["globals"]
+		globals["option:render:camera"] = IECore.StringData( "expression" )
+		parent["dot"]["out"]["globals"] = globals
+		"""
+		) )
+
+		history = GafferScene.SceneAlgo.history( script["dot"]["out"]["globals"] )
+		optionHistory = GafferScene.SceneAlgo.optionHistory( history, "render:camera" )
+
+		self.__assertOptionHistory( optionHistory, [], script["dot"]["out"], "render:camera", "expression", 1 )
+		self.__assertOptionHistory( optionHistory, [ 0 ], script["dot"]["in"], "render:camera", "test", 1 )
+		self.__assertOptionHistory( optionHistory, [ 0, 0 ], script["options"]["out"], "render:camera", "test", 0 )
+
+	def testHistoryWithExpression( self ) :
+
+		script = Gaffer.ScriptNode()
+
+		script["options"] = GafferScene.StandardOptions()
+		script["options"]["options"]["renderCamera"]["enabled"].setValue( True )
+		script["options"]["options"]["renderCamera"]["value"].setValue( "test" )
+
+		script["dot"] = Gaffer.Dot()
+		script["dot"].setup( script["options"]["out"] )
+		script["dot"]["in"].setInput( script["options"]["out"] )
+
+		script["expression"] = Gaffer.Expression()
+		script["expression"].setExpression( inspect.cleandoc(
+		"""
+		globals = parent["dot"]["in"]["globals"]
+		globals["option:render:camera"] = IECore.StringData( "expression" )
+		parent["dot"]["out"]["globals"] = globals
+		"""
+		) )
+
+		def runTest() :
+
+			history = GafferScene.SceneAlgo.history( script["dot"]["out"]["globals"] )
+
+			for plug, value in [
+				( script["dot"]["out"], "expression" ),
+				( script["dot"]["in"], "test" ),
+				( script["options"]["out"], "test" ),
+			] :
+				self.assertEqual( history.scene, plug )
+				self.assertLessEqual( len( history.predecessors ), 1 )
+				history = history.predecessors[0] if history.predecessors else None
+
+			self.assertIsNone( history )
```

Comment:
This is slightly different to the one above in that it is testing the history directly, rather than the filtered optionHistory so we can't reuse `__assertOptionHistory()` here. Having said that, `value` is entirely unnecessary here, so I've tidied things up a bit so we're just looping over the plugs.